azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusUtils.java

```diff
@@ -0,0 +1,81 @@
+package azkaban.executor.container.watch;
+
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_APP_FAILURE;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_COMPLETED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_DELETED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_INIT_FAILURE;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_READY;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_REQUESTED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_SCHEDULED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_UNEXPECTED;
+import static azkaban.executor.container.watch.AzPodStatus.AZ_POD_UNSET;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Contains routines for validation of {@link AzPodStatus} transitions and any other processing
+ * required for the statuses.
+ */
+public class AzPodStatusUtils {
```

Comment:
Let's rename this class if you want to keep the valid graph walk for statuses here or move it to enum.